irksome/pc.py

```diff
@@ -118,20 +121,23 @@ def form(self, pc, test, trial):
         stage_type = appctx.get("stage_type", None)
         bc_type = appctx.get("bc_type", None)
         splitting = appctx.get("splitting", None)
-        nullspace = appctx.get("nullspace", None)
         v0 = oldF.arguments()[0]
 
         F, bcs = self.getNewForm(pc, u0, v0)
         # which getForm do I need to get?
 
+        # get stages
+        ctx = get_appctx(pc.getDM())
+        w = ctx._x
+
         if stage_type in ("deriv", None):
-            Fnew, w, bcnew, bignsp = \
-                getForm(F, butcher_tableau, t, dt, u0, bcs,
-                        bc_type, splitting, nullspace)
+            Fnew, bcnew = \
+                getForm(F, butcher_tableau, t, dt, u0, w, bcs,
+                        bc_type, splitting)
         elif stage_type == "value":
-            Fnew, _, w, bcnew, bignsp = \
-                getFormStage(F, butcher_tableau, u0, t, dt, bcs,
-                             splitting, nullspace)
+            Fnew, _, w, bcnew = \
+                getFormStage(F, butcher_tableau, t, dt, u0, w, bcs,
+                             splitting)
```

Comment:
Feels like there's a lot of duplication between `RanaBase` and `IRKAuxilliaryOperatorPC`.  Are we able to slim things down a bit here, too?

Reply:
Yes, most likely.